pkg/handler3/handler.go

```diff
@@ -157,6 +193,21 @@ func (o *OpenAPIService) HandleDiscovery(w http.ResponseWriter, r *http.Request)
 	http.ServeContent(w, r, "/openapi/v3", time.Now(), bytes.NewReader(data))
 }
 
+func applyCacheBusting(w http.ResponseWriter, r *http.Request, etag string) {
+	hash := r.URL.Query().Get("hash")
+	// Only apply cache busting on requests where the hash is provided.
+	if hash != "" {
+		// The Vary header is required because the Accept header can change the contents returned. This prevents clients from caching protobuf as JSON and vice versa.
+		w.Header().Set("Vary", "*")
```

Comment:
`Vary: *` literally disables all caching AFAIK, so that's not going to work ;-), you need `Vary: Accept` and maybe a few other.

From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Vary#directives:
> ```
> *
>     Indicates that factors other than request headers influenced the generation of this response. Implies that the response is uncacheable.
```